lib/VM/JSLib/DateUtil.cpp

```diff
@@ -972,15 +1045,42 @@ static double parseESDate(StringView str) {
   else
     return nan;
 
-  // Hour and minute of timezone adjustment.
-  if (it > end - 4)
-    return nan;
-  if (!scanInt(it, it + 2, tzh))
-    return nan;
-  tzh *= sign;
-  if (!scanInt(it, it + 2, tzm))
+  // Timezone MUST be set.
+  if (it == end) {
     return nan;
-  tzm *= sign;
+  }
+  // Hour and minute of timezone adjustment.
+  if (consume(u':') && it == end) {
+    // See if timezone is '-:' or '+:'
+    adjustTZ = true;
+  } else if ((it + 1 < end && *(it + 1) == u':') || (it + 2 < end && *(it + 2) == u':')) {
+    // See if timezone seperated by ':', is like '-07:00' or '-7:0' or '-7:00' or '-07:' or '-7:'
+    if (!scanInt(it, end, tzh)) {
+      return nan;
+    }
+    tzh *= sign;
+    if (!consume(u':')) {
+      return nan;
+    }
+    if (scanInt(it, end, tzm)) {
+      tzm *= sign;
+    }
+  } else {
+    // See if timezone didn't seperate by ':'
+    if (it < end - 2) {
+      // See if timezone is like '0700' or '070'
```

Comment:
Do we really want to accept 3-digit timezone adjustments here? User intent here seems non-obvious - is it a number of minutes, are the first two digits the hour, are the last two digits the minutes?

Reply:
I'll revert this changes.